src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +386,35 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Call Box Representations API to fetch thumbnails of all files in collection.
+     * The thumbnailUrls array in state will be populated with urls to thumbnails.
+     * If a thumbnail could not be found, that item's entry in thumbnailUrls will be null.
+     *
+     * @private
+     * @param {Array<BoxItem>} items - item collection object
+     * @param {string} dimensions - desired dimensions of thumbnail. Acceptable dimensions
+     * for a jpg are: "32x32", "94x94", "160x160", "320x320", "1024x1024", "2048x2048".
+     * @return {void}
+     */
+    fetchThumbnailUrls(items: Array<BoxItem>, dimensions: string): void {
+        const fileAPI = this.api.getFileAPI();
+
+        // not using Promise.all since thumbnails should load one at a time
+        items.forEach((item, index) => {
+            new Promise(resolve => {
+                // getFileThumbnail resolves with thumbnailUrl if one is found, null otherwise.
+                fileAPI.getFileThumbnail(item, dimensions, resolve, this.errorCallback);
+            }).then(thumbnailUrl => {
+                const currentCollection = { ...this.state.currentCollection };
```

Comment:
This makes a shallow copy of the currentCollection but the underlying items array is sharing the reference with state.currentCollection so this modification is still a mutation. This is a helpful reference for updating immutable state https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns